test/unit/db/repos/blocks.js

```diff
@@ -0,0 +1,885 @@
+				expect(data[0]).to.have.all.keys(blockListFields);
+
+				// As we only see 5 blocks
+				return expect(data[0].b_height).to.be.eql(5);
+			});
+		});
+
+		describe('loadLastNBlockIds()', () => {
+			it('should use the correct SQL file with one parameter', function*() {
+				sinonSandbox.spy(db, 'query');
+
+				yield db.blocks.loadLastNBlockIds(1);
+
+				expect(db.query.firstCall.args[0]).to.eql(sql.loadLastNBlockIds);
+				expect(db.query.firstCall.args[1]).to.eql([1]);
+				return expect(db.query).to.be.calledOnce;
+			});
+
+			it('should return last block ids', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const data = yield db.blocks.loadLastNBlockIds(5);
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('array');
+				expect(data).to.have.lengthOf(5); // As look for 5 blocks
+				expect(data[0]).to.have.all.keys('id');
+
+				// Due to descending order on height genesis block will appear in last
+				return expect(data[data.length - 1].id).to.be.eql(genesisBlock.id);
+			});
+		});
+
+		describe('blockExists()', () => {
+			it('should use the correct SQL file with one parameter', function*() {
+				sinonSandbox.spy(db, 'oneOrNone');
+
+				yield db.blocks.blockExists('1111');
+
+				expect(db.oneOrNone.firstCall.args[0]).to.eql(sql.blockExists);
+				expect(db.oneOrNone.firstCall.args[1]).to.eql(['1111']);
+				return expect(db.oneOrNone).to.be.calledOnce;
+			});
+
+			it('should return block id if provided id exists', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const data = yield db.blocks.blockExists(genesisBlock.id);
+
+				expect(data).to.be.not.empty;
+				expect(data).to.be.an('object');
+				expect(data).to.have.all.keys('id');
+				return expect(data.id).to.be.eql(genesisBlock.id);
+			});
+
+			it('should return empty response if provided id does not exists', function*() {
+				const data = yield db.blocks.blockExists('111111');
+
+				return expect(data).to.be.null;
+			});
+		});
+
+		describe('deleteAfterBlock()', () => {
+			it('should use the correct SQL file with one parameter', function*() {
+				sinonSandbox.spy(db, 'none');
+
+				yield db.blocks.deleteAfterBlock('1111');
+
+				expect(db.none.firstCall.args[0]).to.eql(sql.deleteAfterBlock);
+				expect(db.none.firstCall.args[1]).to.eql(['1111']);
+				return expect(db.none).to.be.calledOnce;
+			});
+
+			it('should delete all blocks having height above given block id', function*() {
+				const genesisBlock = blocksFixtures.GenesisBlock();
+				const before = yield db.query(
+					'SELECT height FROM blocks WHERE height >= $1',
+					[genesisBlock.height]
+				);
+				const data = yield db.blocks.deleteAfterBlock(genesisBlock.id);
+				const after = yield db.query(
+					'SELECT height FROM blocks WHERE height >= $1',
+					[genesisBlock.height]
+				);
+
+				expect(data).to.be.null;
+				expect(before).to.have.length(5);
+				return expect(after).to.have.length(0);
```

Comment:
Something is wrong, it should not delete genesis block, we should expect `1` here

Reply:
Its doing as expected. I updated the test description accordingly.

```
WHERE height >= (SELECT height FROM blocks WHERE id = $1)
```

This is the condition, so genesis block will also be deleted due to `>=` condition.

Reply:
Yep, looks like function name is misleading.